Transmission.xcodeproj/project.pbxproj

```diff
@@ -3079,19 +3080,20 @@
 			shellPath = /bin/bash;
 			shellScript = "cd third-party/libevent/include/event2\n\nif [ ! -e event-config.h -a ! ../../../macosx-libevent-event-config.h -ef event-config.h ]; then\n    ln -s ../../../macosx-libevent-event-config.h event-config.h;\nfi\n\nif [ ! -e ../evconfig-private.h -a ! ../../macosx-libevent-evconfig-private.h -ef ../evconfig-private.h ]; then\n    ln -s ../../macosx-libevent-evconfig-private.h ../evconfig-private.h;\nfi\n";
 		};
-		C12F197C1E1AE55A0005E93F /* ShellScript */ = {
+		C12F197C1E1AE55A0005E93F /* symlinks */ = {
 			isa = PBXShellScriptBuildPhase;
 			buildActionMask = 2147483647;
 			files = (
 			);
 			inputPaths = (
 			);
+			name = symlinks;
 			outputPaths = (
-				"third-party/miniupnpc/miniupnp",
+				"third-party/miniupnp/miniupnp",
 			);
 			runOnlyForDeploymentPostprocessing = 0;
 			shellPath = /bin/sh;
-			shellScript = "cd third-party/miniupnp && rm -f miniupnp && ln -s . miniupnp\n";
+			shellScript = "cd third-party/miniupnp/miniupnpc\nrm -rf miniupnpc\nmkdir miniupnpc\nln -s ../include miniupnpc/miniupnpc\n";
```

Comment:
If you're changing the paths to the link, then you need to change the paths to the inputs/outputs.

```
diff --git a/Transmission.xcodeproj/project.pbxproj b/Transmission.xcodeproj/project.pbxproj
index e775f18be..1ca9d2d2a 100644
--- a/Transmission.xcodeproj/project.pbxproj
+++ b/Transmission.xcodeproj/project.pbxproj
@@ -3086,10 +3086,11 @@
 			files = (
 			);
 			inputPaths = (
+				"third-party/miniupnp/include",
 			);
 			name = symlinks;
 			outputPaths = (
-				"third-party/miniupnp/miniupnp",
+				"third-party/miniupnp/miniupnpc/miniupnpc/miniupnpc",
 			);
 			runOnlyForDeploymentPostprocessing = 0;
 			shellPath = /bin/sh;
```

Reply:
Also, @nevack, I think there is one too many "miniupnpc". I would change:
```
ln -s ../include miniupnpc/miniupnpc
```
with:
```
ln -s ../include miniupnpc/include
```
(and change the outputPaths accordingly)

Reply:
@Coeur Feel free to try in yourself.
Xcode expects structure like this because the header is included like
`#include <miniupnpc/miniupnpc.h>`.

Reply:
Well, we need at least my patch from the first comment here, for Xcode to do the correct job at knowing regarding outputPaths if the script should run or not.

Reply:
> Well, we need at least my patch from the first comment here, for Xcode to do the correct job at knowing regarding outputPaths if the script should run or not.

654c293

Reply:
I'd say
```diff
...
 			outputPaths = (
-				"third-party/miniupnp/miniupnp",
+				"third-party/miniupnp/miniupnpc/include/miniupnpc",
 			);
 			runOnlyForDeploymentPostprocessing = 0;
 			shellPath = /bin/sh;
-			shellScript = "cd third-party/miniupnp && rm -f miniupnp && ln -s . miniupnp\n";
+			shellScript = "cd third-party/miniupnp/miniupnpc\nln -sf . include/miniupnpc\n";
...
-					"third-party/miniupnpc/*.h",
+					"third-party/miniupnp/miniupnpc/include",
...
```

Reply:
9fce4be